src/Builder.php

```diff
@@ -56,7 +57,7 @@ public static function build()
 
     protected function getPrinter ()
     {
-        $printer = new Printer($this->width, $this->height, $this->fontPath);
+        $printer = new Printer($this->width, $this->height, $this->forceWidth, $this->barcodeLabel, $this->fontPath);

```

Comment:
I think determining the ultimate width is not the responsibility of the Printer itself. Builder should calculate width based on certain conditions. It should be transparent to printer itself imo. 

When it comes to barcodeLabel, Printer constructor does need such information probably. There is another output() method that requires data to encode. It looks like a better place to put additional (optional parameter) to override label (set empty or thoroughly different). This of course should take into consideration the height of the bars (bars may be higher without label without exceeding label height)

Reply:
Just working around current param structure.

Reply:
Printers should not be aware of any width caltulations required to be done. If you introduce another policy to calculate width you will cause ripple effect uncecessarly.  And IMO barcodeLabel is not the printer parameter (more about that below)